protocols/dcutr/src/behaviour_impl.rs

```diff
@@ -329,24 +326,17 @@ impl NetworkBehaviour for Behaviour {
         };
 
         match handler_event {
-            Either::Left(handler::relayed::Event::InboundConnectRequest {
-                inbound_connect,
-                remote_addr,
-            }) => {
-                self.queued_events.extend([
-                    ToSwarm::NotifyHandler {
-                        handler: NotifyHandler::One(relayed_connection_id),
-                        peer_id: event_source,
-                        event: Either::Left(handler::relayed::Command::AcceptInboundConnect {
-                            inbound_connect,
-                            obs_addrs: self.observed_addresses(),
-                        }),
-                    },
-                    ToSwarm::GenerateEvent(Event::RemoteInitiatedDirectConnectionUpgrade {
+            Either::Left(handler::relayed::Event::InboundConnectRequest) => {
+                self.queued_events.extend([ToSwarm::GenerateEvent(
+                    Event::RemoteInitiatedDirectConnectionUpgrade {
                         remote_peer_id: event_source,
-                        remote_relayed_addr: remote_addr,
```

Comment:
Why no longer provide the `remote_relayed_addr` with the `RemoteInitiatedDirectConnectionUpgrade` event? That would remove the need for explicit state tracking in the `NetworkBehaviour` implementation and instead store the state close to its source, namely the connection.

Reply:
Friendly ping. Am I missing something? This should eliminate the necessity for maintaining the state of Behaviour::remote_relayed_addr, correct?

Reply:
Yes-ish. The original idea was to not pass data back and forth. The behaviour learns about the address first so it is kind of redundant to pass it to the handler only to then pass it to the behaviour again.

On the flipside, not needing the hashmap removes a few error cases and simplifies the behaviour so I think it is a good idea.

@tcoratger Mind taking a look at this?

The idea would be to remove the `peer_addresses` hashmap and instead pass the address in the `InboundConnectRequest` enum from the handler to the behaviour.